### DISCUSSION TITLE
Creacion de rama repo remoto git, add, commit, pull, push

### DISCUSSION THREAD

**Discussion by Jose714** [SarayGal/Unac-Virtual-Tracking#63] [Libro de Conocimiento]
Este es un tutorial de como utilizar git en nuestro proyecto este es el link del video: https://youtu.be/0cMPKdv9DaA

**Comment by Nix0n98** [comment]
PARA MACOS: 

1.  Dirígete a la página web de GitHub Desktop: https://desktop.github.com
2. Dale clic en el botón y a continuación se descargará tu app. Da clic en: download for MacOS 
3. Buscamos en nuestro Finder  el archivo que acabamos de descargar para que se ejecute, damos clic y se abre una ventana que nos dice: Instalador de Github Desktop.
4. Hacemos click en: "Continuar"
5. Luego aparece en pantalla el contrato de licencia y hacemos click en: "Continuar"
6. Nos dice que para "continuar con la instalación debemos aceptar las condiciones" y hacemos click en: "Acepto"
7. Luego nos pregunta el destino a instalar y nos muestra nuestro disco duro donde se va instalar y hacemos click en: "Continuar"
8. Ahora nos dice que se hará una "instalación estándar" (por si queremos cambiar el directorio donde se instalará) y hacemos click en: "Instalar"
9. El sistema operativo detecta que se intenta instalar una "software nuevo" por lo cual te pide tu contraseña de la computadora, escribimos la contraseña y continuamos.
10. En este momento inicia la instalación y posteriormente indica que "La instalación se completó correctamente".
11. Luego de finalizar la instalación abrimos la app. 
12. Posteriormente vamos a iniciar sesión con nuestro usuario de GitHub y nos podremos loquear con dos opciones: 
a. Por web
b. Por Github Enterprise. 
13. Cuando ya hayamos seleccionado alguna opción nos saldrá la siguiente pantalla y le damos en "Finish". 
![Captura de pantalla 2021-04-19 a las 21 44 08](https://user-images.githubusercontent.com/80726964/115329903-a0afc580-a158-11eb-8905-e9d2d2d96f8d.png)
14. Posteriormente nos saldrán las opciones para poder duplicar los repositorios o crear uno nuevo. 
![Captura de pantalla 2021-04-19 a las 21 46 19](https://user-images.githubusercontent.com/80726964/115329992-cf2da080-a158-11eb-80b1-2b38c8930c45.png)
*Si ya tenemos un repositorio creado en la web, elegimos la opción "Clone a repositorio from the Internet". 
15. Nos saldrá una pequeña ventana en la cual seleccionaremos el repositorio que tenemos creado y en la parte inferior nos saldrá una opción que dice: "Local Path". Ahí mismo le daremos en el botón "Choose".
16. Ahí buscaremos la carpeta donde vamos a clonar nuestro repositorio. Cabe resaltar que tendrá que ser una carpeta vacía. 
![Captura de pantalla 2021-04-19 a las 21 48 40](https://user-images.githubusercontent.com/80726964/115330334-73174c00-a159-11eb-9cd4-bb545182599d.png)
17. Después le damos en el botón: "Clone".
18. Después de dar clic en el botón "Clone" nos saldrá una nueva ventana donde veremos en la parte superior las ramas que tenemos actualmente. 
![Captura de pantalla 2021-04-19 a las 21 52 37](https://user-images.githubusercontent.com/80726964/115330459-b5408d80-a159-11eb-9a5a-1b2b1724985a.png)
19. Vamos a seleccionar la rama donde vamos a trabajar o podremos crear una nueva rama. 
![Captura de pantalla 2021-04-19 a las 21 55 01](https://user-images.githubusercontent.com/80726964/115330564-ec16a380-a159-11eb-824c-4e54ab67b5d5.png)
20. Posteriormente nos desplazaremos a nuestro Finder y buscaremos la carpeta donde clonamos el repositorio y colocaremos ahí nuestro proyecto. Automáticamente si abrimos nuestra app, nos aparecerán ya los archivos para realizar nuestro commit. 
21. En la parte inferior cerca a nuestra imagen de perfil le colocaremos el nombre a nuestro commit y luego llenamos la descripción. Posteriormente damos clic donde dice: commit front-registro y automáticamente tendremos realizado nuestro commit.
22. Después de realizar el punto 21 le damos en el botón: "fetch origin" y ya quedará automáticamente realizado nuestro push al repositorio remoto en GitHub.com.
![Captura de pantalla 2021-04-19 a las 22 00 42](https://user-images.githubusercontent.com/80726964/115331115-f2f1e600-a15a-11eb-91eb-d49ef6120a30.png)

Y listo, ya realizaste todo lo que necesitabas para hacer tu commit desde MacOS.
